terraform/jenkins-agents-images/main.tf

```diff
@@ -0,0 +1,43 @@
+variable "terraform-images" {
+  type = map
+  default = {
+    "acc-ubuntu-18.04" = {
+      "os_type"  = "Linux"
+      "blob_uri" = "https://oejenkinswesteurope.blob.core.windows.net/disks/jenkins-terraform-ubuntu-18.04.vhd"
+    }
+    "acc-ubuntu-16.04" = {
+      "os_type"  = "Linux"
+      "blob_uri" = "https://oejenkinswesteurope.blob.core.windows.net/disks/jenkins-terraform-ubuntu-16.04.vhd"
+    }
+    "ubuntu-nonSGX" = {
+      "os_type"  = "Linux"
+      "blob_uri" = "https://oejenkinswesteurope.blob.core.windows.net/disks/jenkins-terraform-ubuntu-nonSGX.vhd"
+    }
+    "acc-win-2016-SGX" = {
+      "os_type"  = "Windows"
+      "blob_uri" = "https://oejenkinswesteurope.blob.core.windows.net/disks/jenkins-terraform-win-2016-SGX.vhd"
+    }
+    "acc-win-2016-dcap" = {
+      "os_type"  = "Windows"
+      "blob_uri" = "https://oejenkinswesteurope.blob.core.windows.net/disks/jenkins-terraform-win-2016-dcap.vhd"
+    }
+    "win-2016-nonSGX" = {
+      "os_type"  = "Windows"
+      "blob_uri" = "https://oejenkinswesteurope.blob.core.windows.net/disks/jenkins-terraform-win-2016-nonSGX.vhd"
+    }
+  }
+}
+
+resource "azurerm_image" "terraform-image" {
+  for_each            = var.terraform-images
+  name                = each.key
+  location            = "westeurope"
```

Comment:
I want to say "we should abstract the location".
But given hardware restrictions, this makes sense to keep it locked into WE.